shell/common/persistent_cache_unittests.cc

```diff
@@ -132,6 +143,80 @@ TEST_F(ShellTest, CacheSkSLWorks) {
   DestroyShell(std::move(shell));
 }
 
+TEST_F(PersistentCacheTest, CanPrecompileMetalShaders) {
+#if !SHELL_ENABLE_METAL
+  GTEST_SKIP();
+#endif  //  !SHELL_ENABLE_METAL
+  fml::ScopedTemporaryDirectory dir;
+  PersistentCache::SetCacheDirectoryPath(dir.path());
+  PersistentCache::ResetCacheForProcess();
+
+  auto settings = CreateSettingsForFixture();
+  settings.cache_sksl = true;
+  settings.dump_skp_on_shader_compilation = true;
+
+  fml::AutoResetWaitableEvent first_frame_latch;
+  settings.frame_rasterized_callback =
+      [&first_frame_latch](const FrameTiming& t) {
+        first_frame_latch.Signal();
+      };
+
+  auto sksl_config = RunConfiguration::InferFromSettings(settings);
+  sksl_config.SetEntrypoint("emptyMain");
+  std::unique_ptr<Shell> shell =
+      CreateShell(settings,                                          //
+                  GetTaskRunnersForFixture(),                        //
+                  false,                                             //
+                  nullptr,                                           //
+                  false,                                             //
+                  ShellTestPlatformView::BackendType::kMetalBackend  //
+      );
+  PlatformViewNotifyCreated(shell.get());
+  RunEngine(shell.get(), std::move(sksl_config));
+
+  // Initially, we should have no SkSL cache
+  {
+    auto empty_cache = PersistentCache::GetCacheForProcess()->LoadSkSLs();
+    ASSERT_EQ(empty_cache.size(), 0u);
+  }
+
+  // Draw something to trigger shader compilations.
+  LayerTreeBuilder builder = [](std::shared_ptr<ContainerLayer> root) {
+    SkPath path;
+    path.addCircle(50, 50, 20);
+    auto physical_shape_layer = std::make_shared<PhysicalShapeLayer>(
+        SK_ColorRED, SK_ColorBLUE, 1.0f, path, Clip::antiAlias);
+    root->Add(physical_shape_layer);
+  };
+  PumpOneFrame(shell.get(), 100, 100, builder);
+  first_frame_latch.Wait();
+  WaitForRaster(shell.get());
+  WaitForIO(shell.get());
+
+  // Assert that SKSLs have been generated.
+  auto filled_cache = PersistentCache::GetCacheForProcess()->LoadSkSLs();
+  ASSERT_GT(filled_cache.size(), 0u);
+
+  // Remove all files generated
+  fml::FileVisitor remove_visitor = [&remove_visitor](
+                                        const fml::UniqueFD& directory,
+                                        const std::string& filename) {
+    if (fml::IsDirectory(directory, filename.c_str())) {
+      {  // To trigger fml::~UniqueFD before fml::UnlinkDirectory
+        fml::UniqueFD sub_dir =
+            fml::OpenDirectoryReadOnly(directory, filename.c_str());
+        fml::VisitFiles(sub_dir, remove_visitor);
+      }
+      fml::UnlinkDirectory(directory, filename.c_str());
+    } else {
+      fml::UnlinkFile(directory, filename.c_str());
+    }
+    return true;
+  };
+  fml::VisitFiles(dir.fd(), remove_visitor);
```

Comment:
Can `fml::RemoveFilesInDirectory` be used here?

Reply:
Yeah, you're right. I just copied the test right above this one which was probably written before the utility was available. Updated both.